### PR TITLE
feat: add C implementation for `math/base/special/erfinv`

### LINKED ISSUE #721: [RFC]: Add C implementation for `@stdlib/math/base/special/erfinv`
### Description

This RFC proposes adding C implementation for [`@stdlib/math/base/special/erfinv`](https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/erfinv).

```c
double stdlib_base_erfinv( const double x );
```

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Pranavchiku**
Resolves #721.

## Checklist

- [x] update readme.md
- [x] include.gypi
- [x] binding.gyp
- [x] include/stdlib/math/base/special/
- [x] evalrational.js
- [x] src
- [x] manifest.json
- [x] lib
- [x] examples
- [x] benchmark
- [x] test
* * *

@kgryte

### CODE REVIEW

**Comment by Pranavchiku** [comment]
All tests and benchmarks cleared, this PR can have a final review.

**Comment by Pranavchiku** [comment]
Latest implementation [`1_81_0/boost/math/special_functions/detail/erf_inv.hpp`](https://www.boost.org/doc/libs/1_81_0/boost/math/special_functions/detail/erf_inv.hpp), I tried to have a look at it, but its quite confusing.

**Comment by kgryte** [comment]
@Pranavchiku Indeed, Boost makes heavy use to templating which can be hard to grok. With practice. :)

File: lib/node_modules/@stdlib/math/base/special/erfinv/examples/c/example.c [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/math/base/special/erfinv.h"
+#include <stdlib.h>
+#include <stdio.h>
+
+int main() {
+    double x[] = { -1.0, -0.78, -0.56, -0.33, -0.11, 0.11, 0.33, 0.56, 0.78, 1.0 };
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/erfinv/examples/c/example.c]:
```suggestion
    const double x[] = { -1.0, -0.78, -0.56, -0.33, -0.11, 0.11, 0.33, 0.56, 0.78, 1.0 };
```

File: lib/node_modules/@stdlib/math/base/special/erfinv/README.md [hunk 2]
```diff
@@ -123,6 +123,95 @@ for ( i = 0; i < x.length; i++ ) {
...
+
+### Examples
+
+```c
+#include "stdlib/math/base/special/erfinv.h"
+#include <stdlib.h>
+#include <stdio.h>
+
+int main() {
+    double x[] = { -1.0, -0.78, -0.56, -0.33, -0.11, 0.11, 0.33, 0.56, 0.78, 1.0 };
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/erfinv/README.md]:
```suggestion
    const double x[] = { -1.0, -0.78, -0.56, -0.33, -0.11, 0.11, 0.33, 0.56, 0.78, 1.0 };
```

File: lib/node_modules/@stdlib/math/base/special/erfinv/README.md [hunk 3]
```diff
@@ -123,6 +123,95 @@ for ( i = 0; i < x.length; i++ ) {
...
+    double v;
+    int i;
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/erfinv/README.md]:
```suggestion

```

File: lib/node_modules/@stdlib/math/base/special/erfinv/examples/c/example.c [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/math/base/special/erfinv.h"
+#include <stdlib.h>
+#include <stdio.h>
+
+int main() {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/erfinv/examples/c/example.c:23]: This should be TAB indentation.

Comment (Planeshifter) [reply lib/node_modules/@stdlib/math/base/special/erfinv/examples/c/example.c:23]: Addressed.

**Review by Planeshifter — APPROVED** [review]
Let's get this in!

**Pull request merged**